File: 1814-count-nice-pairs-in-an-array/1814-count-nice-pairs-in-an-array.java
import java.util.HashMap;
import java.util.Map;
class Solution {
    public int countNicePairs(int[] nums) {
        Map<Integer, Integer> arr = new HashMap<>();
        for (int i : nums) {
            int temp = i - reverse(i);
            arr.put(temp, arr.getOrDefault(temp, 0) + 1);
        }
        long res = 0;
        int mod = 1000000007;
        for (int i : arr.values()) {
            res = (res % mod + ((long) i * (i - 1) / 2) % mod) % mod;
        }
        return (int) res;
    }
    private int reverse(int num) {
        int rev = 0;
        while (num > 0) {
            rev = rev * 10 + num % 10;
            num /= 10;
        }
        return rev;
    }
}

File: 1814-count-nice-pairs-in-an-array/1814-count-nice-pairs-in-an-array.py
class Solution:
    def countNicePairs(self, nums):
        arr = {}
        for i in nums:
            temp = i - self.reverse(i)
            if temp in arr:
                arr[temp] += 1
            else:
                arr[temp] = 1
        res = 0
        mod = 1000000007
        for i in arr.values():
            res = (res % mod + (i * (i - 1) // 2)) % mod

        return int(res)

    def reverse(self, arr):
        rev = 0
        while arr > 0:
            rev = rev * 10 + arr % 10
            arr //= 10
        return rev